acmeproxy/proxy/tests/test_views.py

```diff
@@ -0,0 +1,324 @@
+        Authorisation.objects.create(
+            name="example.com", secret="test_secret", created_by_ip="127.0.0.1"
+        )
+        resp = client.post(
+            "/publish_response", data={"name": "example.com", "secret": "test_secret"},
+        )
+        assert resp.status_code == 400
+
+    def test_publish_missing_secret(self, client):
+        Authorisation.objects.create(
+            name="example.com", secret="test_secret", created_by_ip="127.0.0.1"
+        )
+        resp = client.post(
+            "/publish_response",
+            data={"name": "example.com", "response": "random_secret"},
+        )
+        assert resp.status_code == 400
+
+    def test_expire_response(self, client):
+        Authorisation.objects.create(
+            name="example.com", secret="test_secret", created_by_ip="127.0.0.1"
+        )
+        Response.objects.create(
+            name="example.com", response="test_response", created_by_ip="127.0.0.1"
+        )
+        resp = client.post(
+            "/expire_response", data={"name": "example.com", "secret": "test_secret"},
+        )
+        assert resp.status_code == 200
+        assert resp.json()["result"]["authorisation"] == "example.com"
+
+    def test_expire_response_mixed_case(self, client):
+        Authorisation.objects.create(
+            name="example.com", secret="test_secret", created_by_ip="127.0.0.1"
+        )
+        Response.objects.create(
+            name="example.com", response="test_response", created_by_ip="127.0.0.1"
+        )
+        resp = client.post(
+            "/expire_response", data={"name": "exAMple.Com", "secret": "test_secret"},
+        )
+        assert resp.status_code == 200
+        assert resp.json()["result"]["authorisation"] == "example.com"
+
+    def test_expire_missing_name(self, client):
+        Authorisation.objects.create(
+            name="example.com", secret="test_secret", created_by_ip="127.0.0.1"
+        )
+        Response.objects.create(
+            name="example.com", response="test_response", created_by_ip="127.0.0.1"
+        )
+        resp = client.post("/expire_response", data={"secret": "test_secret"},)
+        assert resp.status_code == 400
+
+    def test_expire_missing_secret(self, client):
+        Authorisation.objects.create(
+            name="example.com", secret="test_secret", created_by_ip="127.0.0.1"
+        )
+        Response.objects.create(
+            name="example.com", response="test_response", created_by_ip="127.0.0.1"
+        )
+        resp = client.post("/expire_response", data={"name": "example.com"},)
+        assert resp.status_code == 400
+
+    def test_expire_wrong_secret(self, client):
+        Authorisation.objects.create(
+            name="example.com", secret="test_secret", created_by_ip="127.0.0.1"
+        )
+        Response.objects.create(
+            name="example.com", response="test_response", created_by_ip="127.0.0.1"
+        )
```

Comment:
Like in an earlier comment, this pattern seems to be repeated a lot. Its tricky, because removing this duplication doesn't really remove any lines of code, but perhaps it could make it more clear what information is relevant for the test.